_posts/2019-07-31-dask-2.2.md

```diff
@@ -0,0 +1,343 @@
+and ~~hdfs3~~ [pyarrow.hdfs](https://arrow.apache.org/docs/python/filesystems.html),
+it's easy to read and write data in a Pythonic way to a
+variety of remote storage systems.
+
+Early on we decided that we wanted this code to live outside of the mainline
+Dask codebase, which is why they are independent projects.
+This choice allowed other libraries, like Pandas, Zarr, and others to benefit
+from this work, without a strict dependency on Dask.
+However, there was still code within Dask that helped to unify them a bit.
+We've moved this code out to an external project,
+[fsspec](https://filesystem-spec.readthedocs.io/en/latest) which includes all
+of the centralization code that Dask used to provide, as well as a formal
+specification for what a remote data system should look like in order to be
+compatible.  This also helps to unify efforts with other projects like Arrow.
+
+Special thanks to [Martin Durant](https://github.com/martindurant) for
+shepharding Dask's I/O infrastructure over the years, and for doing the more
+immediate work of splitting out `fsspec`.
+
+
+Async/Await everywhere, and cleaner setup for developers
+--------------------------------------------------------
+
+In Dask 2.0 we dropped Python 2 support and now support only Python 3.5.
+This allows us to adopt async and await syntax for concurrent execution rather
+than an older coroutine based approach with `yield`.  The differences here
+started out as largely aesthetic, but triggered a number of substantive
+improvements as we walked through the codebase cleaning things up.  Starting
+and stopping internal Scheduler, Worker, Nanny, and Client objects is now far
+more uniform, reducing the presence of subtle bugs.
+
+This is discussed in more detail in the [Python API setup
+documentation](https://docs.dask.org/en/latest/setup/python-advanced.html) and
+is encapsulated in this code example from those docs:
+
+```python
+import asyncio
+
+from dask.distributed import Scheduler, Worker, Client
+
+async def f():
+    async with Scheduler() as s:
+        async with Worker(s.address) as w1, Worker(s.address) as w2:
+            async with Client(s.address, asynchronous=True) as client:
+                future = client.submit(lambda x: x + 1, 10)
+                result = await future
+                print(result)
+
+asyncio.get_event_loop().run_until_complete(f())
+```
+
+As a result of this and other internal cleanup intermittent testing failures in
+our CI have disappeared, and developer mood is high :)
+
+
+SSHCluster, a second approach
+-----------------------------
+
+We've added a second SSH cluster deployment solution.  It looks like this:
+
+```python
+from distributed.deploy.ssh2 import SSHCluster  # this will move in future releases
+
+cluster = SSHCluster(
+    hosts=["localhost", "localhost", "localhost"],
+    worker_kwargs={"nthreads": 4},
+    scheduler_kwargs={},
+    connect_kwargs={"known_hosts": None}
```

Comment:
I'd use `SSHCluster` with different hosts if I used it. I'm curious on how to do that, but this example doesn't seem to reflect that.